webapp/channels/src/packages/mattermost-redux/src/actions/posts.ts

```diff
@@ -1112,10 +1115,19 @@ export function getProfilesAndStatusesForPosts(postsArrayOrMap: Post[]|PostList[
     }
 
     // Profiles of users mentioned in the posts
-    const usernamesToLoad = getNeededAtMentionedUsernames(state, postsArray);
+    const usernamesAndGroupsToLoad = getNeededAtMentionedUsernamesAndGroups(state, postsArray);
 
-    if (usernamesToLoad.size > 0) {
-        promises.push(getProfilesByUsernames(Array.from(usernamesToLoad))(dispatch, getState));
+    if (usernamesAndGroupsToLoad.size > 0) {
+        // We need to load the profiles synchronously to filter them
+        // out of the groups to check
+        const getProfilesPromise = getProfilesByUsernames(Array.from(usernamesAndGroupsToLoad))(dispatch, getState);
+        promises.push(getProfilesPromise);
+
+        const {data} = await getProfilesPromise as ActionResult<UserProfile[]>;
+        const loadedProfiles = (data || []).map((p) => p.username);
+        const groupsToCheck = Array.from(usernamesAndGroupsToLoad).filter((name) => !loadedProfiles.includes(name));
```

Comment:
O(n2) (a complete run over the array, and for each element, a complete run on loadedProfiles)

Using a set should get this to O(n).

Reply:
Fixed in https://github.com/mattermost/mattermost/pull/24292/commits/7c3fc6f78b605497096056f3bca06cacd6df2fb9, PTAL